wordpress-mobile/WordPress-iOS
# Site Design Revamp - Track Recommended Designs

Fixes #18582

This pr adds a "recommended" field to the template selection event (fired when users choose a template to create a site), in order to track if the template comes from the recommended list. the field will be set to `true` (`1`) if that is the case, `false` (`0`) otherwise.

To test:

- checkout/build/run this branch
- start the site creation flow
- select a design from the recommended list on top
- tap "Choose"
- Make sure you see an event in the console that contains the value `recommended: 1` like the one below

`🔵 Tracked: enhanced_site_creation_site_design_selected <default_tab_experiment: dashboard, recommended: 1, template: blog>`

- repeat the site creation flow, and select a design that is not in the recommended list on top
-  Make sure you see an event in the console that contains the value `recommended: 0` like the one below

`🔵 Tracked: enhanced_site_creation_site_design_selected <default_tab_experiment: dashboard, recommended: 0, template: links>`

## Regression Notes
1. Potential unintended areas of impact
none, this just adds a field in one analytics event

2. What I did to test those areas of impact (or what existing automated tests I relied on)
make sure the event is tracked correctly

3. What automated tests I added (or what prevented me from doing so)
none
PR submission checklist:
